### ISSUE TITLE
GLsizei is read as Lsizei skipping the first G

### ISSUE THREAD

**Issue by Raffaello** [streetsidesoftware/vscode-spell-checker#2997]
see screenshot:

![image](https://github.com/streetsidesoftware/cspell-dicts/assets/1850902/cd459c57-274d-4dbd-9f5e-3aa11e60a232)

**Comment by Jason3S** [comment]
@Raffaello,

When looking at camel case and snake case words, the spell checker reports on the least number of errors with the highest percentage of the word being matched.

For example it is common to see both `ERRORcode` and `ERRORCode`. The spell checker will accept both. When given `ERRORcode` it tries both: `["ERRO", "Rcode"]` and `["ERROR", "code"]`. Since the second option resulting in all words being found in the dictionary, `ERRORcode` is considered correct. But, if the word was `ERRORcodde`, it will report only `codde` as incorrect because it found `ERROR` in the dictionary.   

In the case of `GLsizei` it tries `[GLsizei]`, `[GL,sizei]`, and `[G,Lsizei]`. Since `G` is in the dictionary, it reports `Lsizei` as the error.

Try experimenting with the following:

```cpp
// cspell:words GL
const GLsizei width;
```

```cpp
// cspell:words sizei
const GLsizei width;
```

```cpp
// cspell:words GLsizei
const GLsizei width;
```

**Comment by Raffaello** [comment]
i see. thanks.

i reported because the `GLsizei` is a typedef of OpenGL, so kind of keyword in C/C++ languages.

How can i add the full word in the dictionary?
